aiida/tools/dbimporters/plugins/materialsproject.py

```diff
@@ -0,0 +1,198 @@
+            except KeyError:
+                raise ValueError('API key not supplied and PMG_MAPI_KEY environment '
+                                 'variable not set. Either pass it when initializing the class, '
+                                 'or set the environment variable PMG_MAPI_KEY to your API key.')
+        self._api_key = api_key
+        self._verify_api_key()
+        self._mpr = MPRester(self._api_key)
+
+    def _verify_api_key(self):
+        """
+        Verify the supplied API key by issuing a request to Materials Project.
+        """
+        response = requests.get(
+            'https://www.materialsproject.org/rest/v1/api_check', headers={'X-API-KEY': self._api_key})
+        response_content = response.json()  # a dict
+        if 'error' in response_content:
+            raise RuntimeError(response_content['error'])
+        if not response_content['valid_response']:
+            raise RuntimeError('Materials Project did not give a valid response for the API key check.')
+        if not response_content['api_key_valid']:
+            raise RuntimeError('Your API key for Materials Project is not valid.')
+
+    @property
+    def api_key(self):
+        """
+        Return the API key configured for the importer
+        """
+        return self._api_key
+
+    @property
+    def properties(self):
+        """
+        Return the properties that will be queried
+        """
+        return self._properties
+
+    @property
+    def get_supported_keywords(self):
+        """
+        Returns the list of all supported query keywords
+
+        :return: list of strings
+        """
+        return self._supported_keywords
+
+    def query(self, **kwargs):
+        """
+        Query the database with a given dictionary of query parameters for a given properties
+
+        :param query: a dictionary with the query parameters
+        :param properties: the properties to query
+        """
+        try:
+            query = kwargs['query']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `query` as a key. This '
+                                 'should contain a dictionary with the right query needed.')
+        try:
+            properties = kwargs['properties']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `properties` as a key.')
+
+        if not isinstance(query, dict):
+            raise TypeError('The query argument should be a dictionary')
+
+        if properties is None:
+            properties = self._properties
+
+        if properties != 'structure':
+            raise ValueError('Unsupported properties: {}'.format(properties))
+
+        results = []
+        properties_list = ['material_id', 'cif']
+        for entry in self.find(query, properties_list):
+            results.append(entry)
+        search_results = MaterialsProjectSearchResults(results, return_class=MaterialsProjectCifEntry)
```

Comment:
``return_class=MaterialsProjectCifEntry`` is built-in in ``MaterialsProjectSearchResults`` thus there is no need to pass it.

Reply:
Yes, this was surely unnecessary. However, I kept the pass as we possibly need that if we introduce other things than `structure`. Should be easy for the user to extend. Fixed.